src-v2/NTS.Domain/Objects/Timestamp.cs

```diff
@@ -28,6 +28,11 @@ public Timestamp Add(TimeSpan span)
         return new Timestamp(DateTime.Add(span));
     }
 
+    public Timestamp SetTime(TimeSpan span)
+    {
+        return new Timestamp(System.DateTime.Today.Add(span));
+    }
```

Comment:
I don't like this method. The main idea of the Timestamp is to represent a snapshot in the current time. It doesn't make much sense to Set it's time. But there is a bigger problem. Any idea what it might be?